src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -162,7 +169,23 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
       primaryLabel: {
         type: LabelSignalEnumType,
         description: "Primary label signal available to collector",
-        resolve: (artwork, {}, ctx) => getPrimaryLabel(artwork, ctx),
+        args: {
+          ignore: {
+            type: GraphQLList(LabelSignalEnumType),
+            description: "Signals to ignore",
+          },
+        },
+        resolve: (artwork, args, ctx) => {
+          const { ignore } = args
+          if (ignore?.length > 0) {
+            if (ignore.length > AVAILABLE_LABEL_COUNT) {
+              throw new Error(
+                `Ignore list length limited to number of available signals - max ${AVAILABLE_LABEL_COUNT}`
+              )
+            }
+          }
```

Comment:
I imagined a long list of ignored labels causing problems for us. Not sure if this is actually necessary. Started with a uniqueness test, now i'm just limiting the total length to the length of possible enum values.